### PR TITLE
Implement visual only newlines, and improve overall newlines handling.

### PR DESCRIPTION
**Pull request by diegoreymendez**
<h3>Description</h3>

- This PR adds visual-only newlines, and makes sure that editing them does not affect the DOM (for the time being)
- Future work must make these visual-only newlines affect the content around them, such as blockquotes being removed when their preceding newline is removed.
- Breaks added while in visual mode, will now be saved as `<br>` tags in the DOM.

<h3>Testing:</h3>

Make sure any issue found is not present in `develop`.  Issues already present in `develop` fall outside of the scope of this PR.

Remember to run the unit tests.  Several were added to try and cover the new logic.

**Test 1:**

1. Launch the populated editor
2. Edit some text (without touching visual-only newlines).
3. Switch to HTML mode
4. Switch to visual mode.
5. Make sure your text edits are still there.

**Test 2:**

1. Launch the empty editor.
2. Switch to HTML mode.
3. Write exactly: `Hello there!<blockquote>This is a blockquote</blockquote>`
4. Switch to visual mode.
5. Make sure there's a newline between the regular text and the blockquote.
6. Switching back and forth should keep HTML mode without the break and visual mode with it.
7. While in visual mode, remove the newline (this will still not affect the blockquote in HTML).
8. Switch to HTML.
9. Switch to visual.
10. The visual-only newline should still be shown, as we're still not removing blockquotes this way.

### CODE REVIEW

**Comment by diegoreymendez** [comment]
Fixes #216

**Comment by diegoreymendez** [comment]
Also fixes #63.

**Comment by diegoreymendez** [comment]
Also fixes #164.

File: Aztec/Classes/Extensions/Character+Name.swift [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    
+    enum Name: Character {
+        case newline = "\n"
+        case space = " "
+        case zeroWidthSpace = "\u{200B}"
+    }
+    
+    init(_ characterName: Name) {
+        self.init(unicodeScalarLiteral: characterName.rawValue)
+    }
```
Comment (jleandroperez) [inline Aztec/Classes/Extensions/Character+Name.swift]: Love this mechanism. Thanks for this!!!

File: Aztec/Classes/Extensions/NSAttributedString+Lists.swift [hunk 2]
```diff
@@ -32,7 +32,7 @@ extension NSAttributedString
         if length > 0 && location > 0 {
             let previousRange = NSRange(location: location - 1, length: 1)
             let previousString = attributedSubstring(from: previousRange)
-            isStartOfLine = previousString.string == "\n"
+            isStartOfLine = previousString.string == String(.newline)
```
Comment (jleandroperez) [inline Aztec/Classes/Extensions/NSAttributedString+Lists.swift]: This is just beautiful code!!!

**Comment by jleandroperez** [comment]
Sorry guys, i'm late to the party. Stunning quality of code sir.

Late :shipit: on my side!!

**Comment by diegoreymendez** [comment]
Ty for the double check, Jorge!

**Pull request merged**